moebot_bot/util/db/types/scheduler.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import "time"
+
+type SchedulerType int
+
+type ScheduledOperation struct {
+	ID                   int64
+	ServerID             int
+	Type                 SchedulerType
+	PlannedExecutionTime time.Time
+}
+
+type ChannelRotation struct {
+	ChannelUIDList    []string
+	CurrentChannelUID string
+	ScheduledOperation
+}
+
+func (c *ChannelRotation) NextChannelUID() string {
```

Comment:
This one is placed in the same file because, while not strictly a type, it's a method of the ChannelRotation struct. Methods of a type have to be placed in the same package of the struct they refer to.

Reply:
Is there a reason we couldn't just have a function `NextChannelUID(c ChannelRotation)`? I realize that's just what a receiver is, but we're not modifying the internal state of this `ChannelRotation` object and I don't see this as a `type`. Perhaps that's just the package name and a more specific one could be used. Regardless, a `types` package seems like an odd place to have functions placed within it.

(Yes I know `raffle` has one too, but we're deleting that anyways)

Reply:
moved the code in the channelRotationScheduler.go file in commit https://github.com/camd67/moebot/pull/67/commits/b132b370a559c725b2780cc9a81fad59a90a5aab